fabfile.py

```diff
@@ -0,0 +1,264 @@
+    if files.exists('/var/run/supervisor.sock'):
+        sudo('unlink /var/run/supervisor.sock')
+    sudo('apt-get install -y '
+         'nginx git python-pip postgresql '
+         'postgresql-contrib libpq-dev python-dev')
+    if not files.exists(
+        'etc/nginx/sites-available/original-default',
+        use_sudo=True
+    ):
+        sudo(
+            'cp /etc/nginx/sites-available/default '
+            '/etc/nginx/sites-available/original-default'
+        )
+
+    put(local_path="~/projects/t-buddies/simple_nginx_conf",
+        remote_path="/etc/nginx/sites-available/default",
+        use_sudo=True)
+
+    append('/etc/nginx/sites-available/default',
+           "server {listen 80;server_name " +
+           env.active_instance.public_dns_name + "/;"
+           "access_log /var/log/nginx/test.log;location /"
+           "{proxy_pass http://127.0.0.1:8080;proxy_set_header Host $host;"
+           "proxy_set_header X-Real-IP $remote_addr;"
+           "proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;"
+           "}}")
+
+    if not files.exists("~/.previous/"):
+        run('mkdir ~/.previous')
+
+    sudo('service nginx start')
+
+
+def _deploy():
+    sudo('service nginx stop')
+
+    from datetime import datetime
+    now = datetime.now()
+    d = now.strftime("%Y_%m_%e__%H_%M_%S")
+
+    if files.exists('~/{p}'.format(p=projname)):
+        run('mv ~/{p} ~/.previous/{d}'.format(p=projname, d=d))
+
+    run(
+        'git clone -b master'
+        ' http://github.com/{gp}'.format(gp=git_path,)
+    )
+
+    if files.exists('/{p}/supervisord.conf'.format(p=projname),
+                    use_sudo=True,):
+        sudo('rm -f /{p}/supervisord.conf'.format(p=projname))
+    sudo('mv {p}/supervisord.conf /etc/supervisord.conf'.format(p=projname))
+
+    sudo('pip install -r ~/{p}/requirements.txt'.format(p=projname))
+    sudo('cd taste-buddies ; python setup.py develop')
+    sudo('reboot')
+    time.sleep(90)
+    sudo('service nginx start')
+    sudo('supervisord')
+
+
+def deploy_app():
+    run_command_on_selected_server(_setup_suite)
+    run_command_on_selected_server(_deploy)
+
+
+# def setup_suite():
+#     run_command_on_selected_server(_setup_suite)
+
+
+# def _install_nginx():
+#     sudo('/etc/init.d/nginx start')
+
+
+def _setup_ngnix():
+    try:
+        sudo('mv /etc/nginx/sites-available/default '
+             '/etc/nginx/sites-available/default.orig')
+    except:
+        pass
+    sudo('mv simple_nginx_conf /etc/nginx/sites-available/default')
+    append('/etc/nginx/sites-available/default',
+           "server {listen 80;server_name " +
+           env.active_instance.public_dns_name + "/;"
+           "access_log /var/log/nginx/test.log;location /"
+           "{proxy_pass http://127.0.0.1:8080;proxy_set_header Host $host;"
+           "proxy_set_header X-Real-IP $remote_addr;"
+           "proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;"
+           "}}")
```

Comment:
notice that you have this code in here twice, which is a sign that there should be a better way to do what you're doing.